larray/core/array.py

```diff
@@ -5644,7 +5644,9 @@ def opmethod(self, other) -> 'Array':
             # TODO: ndarray should probably be converted to larrays because that would harmonize broadcasting rules, but
             # it makes some tests fail for some reason.
             if not isinstance(other, (Array, np.ndarray)) and not np.isscalar(other):
-                if isinstance(other, list):
+                if other is None:
+                    other = full(self.axes, fill_value=None)
+                elif isinstance(other, (list, Axis)):
```

Comment:
using asarray() for the None case should work too I think (it used to work). It should return a "scalar array" (ie an array with 0 axes), which is then broadcasted during the numpy binop, so it should be the same result but computed much more efficiently.

Reply:
Are you sure it is computed much more efficiently with asarray() ? 
It sounds counterintuitive (to me at least).

To me, using full() allow to bypass the broadcasting, so I'm a bit confused.

I'm not saying I disagree but I simply don't understand.

Reply:
That's an excellent question. In fact, you are right that in the case of the current larray version, broadcasting a scalar Array  against a small array is sufficiently expensive that it is more efficient to "expand" the value manually

```python
>>> %timeit arr == asarray(None)
82.5 µs ± 138 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)
>>> %timeit arr == full(arr.axes, fill_value=None)
62.2 µs ± 784 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)
```

However, given this is the exception to the general rule, it is a very rare case (and it's not the case if the other operand is larger), I do not think it is worth it to optimize for this case, so I still think the shorter "other is None or isinstance(other, (list, Axis)):" is better.

Now for the general performance of broadcasting, the binary operations with broadcasting are **faster** than the operations on "full" arrays which have the same axes/shape, except on very small arrays. I suppose this is because memory access is generally more expensive than CPU cycles. The larger the "full arrays", the more benefit you get from broadcasting.

If you want the details, there is some fixed overhead of broadcasting in larray, which means that the broadcasting binop itself is only faster than its full array equivalent binop when the arrays are "largish" **BUT** if you also take into account the time to create the full array from the "smaller" array, the broadcasting op is faster in all but the most extreme cases (like the scalar Array in a binop with a small array above).